### PR TITLE
[browser] cleanup of early startup code

### PR DESCRIPTION
**Pull request by pavelsavara**
Split off from https://github.com/dotnet/runtime/pull/83364
Contributes to https://github.com/dotnet/runtime/issues/72990
Contributes to https://github.com/dotnet/runtime/issues/85045

- drop `dotnet.es6.post.js` 
- simplify `dotnet.es6.lib.js` and `dotnet.es6.extpost.js` and `dotnet.es6.pre.js`
- create `types-api.ts` by moving types from other files
- drop `export_linker()` from startup
- drop `IMPORTS` object
- drop `isGlobal` flag (CJS cleanup)
- reorganized `init_polyfills_async` and `init_replacements`
- refactored `unifyModuleConfig` from `dotnet.es6.pre.js` to `run-outer.ts`
- removed `INTERNAL` from top most export

### CODE REVIEW

**Comment by pavelsavara** [comment]
All the CI failures are unrelated.

File: src/mono/wasm/runtime/types-api.ts [hunk 1]
```diff
@@ -0,0 +1,245 @@
+import type { EmscriptenModule, NativePointer } from "./types/emscripten";
+
+export interface DotnetHostBuilder {
```
Comment (maraf) [inline src/mono/wasm/runtime/types-api.ts:3]: I don't see `withStartupMemoryCache`, it that correct?

**Review by maraf — COMMENTED** [review]
It looks great! I have just some minor questions

File: src/mono/wasm/runtime/types-api.ts (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/types-api.ts:3]: Yes, that's not public API

File: src/mono/wasm/runtime/icu.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 import cwraps from "./cwraps";
-import { Module, runtimeHelpers } from "./imports";
+import { ENVIRONMENT_IS_WEB, Module, runtimeHelpers } from "./imports";
```
[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/mono/wasm/runtime/icu.ts:5]: line 67

File: src/mono/wasm/runtime/imports.ts [hunk 3]
```diff
@@ -43,27 +36,37 @@ export function set_imports_exports(
     disableLegacyJsInterop = imports.disableLegacyJsInterop;
     runtimeHelpers.quit = imports.quit_;
     runtimeHelpers.ExitStatus = imports.ExitStatus;
-    runtimeHelpers.requirePromise = imports.requirePromise;
 }
 
-export function set_emscripten_entrypoint(
-    entrypoint: CreateDotnetRuntimeType
+export function set_exports(
```
[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/mono/wasm/runtime/imports.ts:41]: renamed to `initializeGlobals` and also renamed the file to `globals.ts`

[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/mono/wasm/runtime/imports.ts:41]: It generated large noisy diff, but I think it's good thing to do.

File: src/mono/wasm/runtime/snapshot.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@
 import ProductVersion from "consts:productVersion";
 import GitHash from "consts:gitHash";
 import MonoWasmThreads from "consts:monoWasmThreads";
-import { runtimeHelpers } from "./imports";
+import { ENVIRONMENT_IS_WEB, runtimeHelpers } from "./imports";
```
[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/mono/wasm/runtime/snapshot.ts:7]: line 20

**Review by maraf — APPROVED** [review]
Looks good to me 👍

**Pull request merged**